### ISSUE TITLE
reflog / be able to undo a merge

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#773]
```
  Tip: You can always `undo` if this wasn't what you wanted.

.> undo

  ⚠️
  
  Sorry, I can't undo a merge (not implemented yet).
```
:rage4:

**Comment by aryairani** [comment]
- [ ] codebase can auto-complete branch hashes
- [ ] `stepAt` and the like add an entry to some reflog store when updating the root
- [ ] create `reflog` command that lists and describes those past root updates
```
.> reflog
 Use `reset-root <index/hash>` to set the root namespace to a particular one.
 The `undo` command is shorthand for `reset-root 1`.

   [1] #abcdefg (pull https://asdf.com/asdf.git)
   [2] #hijklmn (update) -- for an add/update, I'd like to show a diff, but a 1-liner would be nice)
   [3] #opqrst (reset-root #
```
- [ ] add `reset-root #hash` command
- [ ] update commands that take a namespace hash to support a unique namespace hash prefix
  - [ ] `history`
  - [ ] `reset-root`

**Comment by aryairani** [comment]
```
.> reflog

  ⚠️
  
  The reflog appears to be empty!

.> ls

  1. boo/     (2 definitions)
  2. builtin/ (265 definitions)

.> move.namespace boo foo

  Done.

.> reflog

  1. #ntceenm7s730o7l5j24908f5qg1930r69c4lo3dkd8rt3tai4il3hmij7vlq5js8k77kjpji955ufkf4mavinfce7qfrk7gl6jgm0pg : move.namespace .boo .foo

.> fork #ntceenm7s730o7l5j24908f5qg1930r69c4lo3dkd8rt3tai4il3hmij7vlq5js8k77kjpji955ufkf4mavinfce7qfrk7gl6jgm0pg oldroot

  Done.

.> ls oldroot

  1. builtin/ (265 definitions)
  2. foo/     (2 definitions)

.> 
```
not bad imo

**Comment by aryairani** [comment]
Implemented in #858
